plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php

```diff
@@ -46,6 +47,12 @@ class CustomOrdersTableController {
 
 	public const DEFAULT_DB_TRANSACTIONS_ISOLATION_LEVEL = 'READ UNCOMMITTED';
 
+	public const HPOS_FTS_INDEX_OPTION = 'woocommerce_hpos_fts_index_enabled';
+
+	public const HPOS_FTS_ADDRESS_INDEX_CREATED_OPTION = 'woocommerce_hpos_address_fts_index_created';
+
+	public const HPOS_FTS_ORDER_ITEM_INDEX_CREATED_OPTION = 'woocommerce_hpos_order_item_fts_index_created';
```

Comment:
Is there any reason to truly have these as separate options from the feature option itself? I mean, if instead of hooking onto `updated_option` we hooked onto `pre_update_option` we could attempt to create the indexes and return `no` if we were not able to do so. That way, if the feature is enabled we can assume the indexes are there. In any case, when the feature is enabled we use `DatabaseUtil` to find whether the indexes exist, so those options are only useful at query time (but then, we could just check the feature option instead).

I'm just in favor of option economy, but do let me know if I'm missing something.

If we later add new indexes as part of this feature, we could handle those with migration routines as usual. wdyt?

Reply:
So the utility of two different options is that I am trying to see if there are failures associated with creating this index on a possibly large table (order items). If there is, then we can evaluate a different approach or drop this one. The search query also uses these options use either the normal or the fts query, per index.

I agree with your point about options economy, but let's remove it when we promote this feature to stable (and no that both indexes are equally robust). wdyt?

Reply:
Sounds good!